src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,125 @@
+
+    @Test
+    public void userWasAddedAndReturned_Ok() {
+        User expected = new User(
+                (long) FIRST_VALID_AGE, THIRD_LOGIN, FIRST_VALID_PASSWORD, FIRST_VALID_AGE);
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual, "User should be returned");
+    }
+
+    @Test
+    public void registerNullUser_NotOk() {
+        assertThrows(InvalidInputException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    public void registerNullAge_NotOk() {
+        firstUser.setAge(null);
+        assertThrows(InvalidInputException.class, () ->
+                        registrationService.register(firstUser),
+                "Method should throw InvalidInputException for null age");
+    }
+
+    @Test
+    public void registerNullLogin_NotOk() {
+        firstUser.setLogin(null);
+        assertThrows(InvalidInputException.class, () ->
+                        registrationService.register(firstUser),
+                "Method should throw InvalidInputException for null login");
+    }
+
+    @Test
+    public void registerNullPassword_NotOk() {
+        firstUser.setPassword(null);
+        assertThrows(InvalidInputException.class, () ->
+                        registrationService.register(firstUser),
+                "Method should throw InvalidInputException for null password");
+    }
+
+    @Test
+    public void registerLoginExists_NotOk() {
+        secondUser.setLogin(FIRST_LOGIN);
+        assertThrows(InvalidInputException.class, () -> registrationService.register(secondUser),
+                "Method should throw InvalidInputException if login exists");
+    }
+
+    @Test
+    public void registerUserExists_NotOk() {
+        assertThrows(InvalidInputException.class, () -> registrationService.register(firstUser),
+                "Method should throw InvalidInputException if user exists");
+    }
+
+    @Test
+    public void registerInvalidUsersAge_NotOk() {
+        firstUser.setAge(INVALID_AGE);
+        assertThrows(InvalidInputException.class, () -> registrationService.register(firstUser),
+                "Method should throw InvalidInputException for invalid age");
+
+    }
+
+    @Test
+    public void registerInvalidUsersPassword_NotOk() {
+        firstUser.setPassword(INVALID_PASSWORD);
+        assertThrows(InvalidInputException.class, () -> registrationService.register(firstUser),
+                "Method should throw InvalidInputException for invalid password");
+    }
+
+    @Test
+    public void getUser_Ok() {
+        User actualFirst = registrationService.getUser(FIRST_LOGIN);
+        User actualSecond = registrationService.getUser(SECOND_LOGIN);
+        assertEquals(firstUser, actualFirst);
+        assertEquals(secondUser, actualSecond);
+    }
+
+    @Test
+    public void getUserNonExistLogin_Ok() {
+        User actual = registrationService.getUser(FIRST_VALID_PASSWORD);
+        assertNull(actual);
+    }
+}
```

Comment:
You should test edgecases with password.length 5/6/7 and age 17/18/19
and test names should be more clear: ...lowerThenExpected  ...equalsExpected  ...higherThenExpected